app/webhook.go

```diff
@@ -602,8 +624,15 @@ func (a *App) HandleIncomingWebhook(hookId string, req *model.IncomingWebhookReq
 		return model.NewAppError("HandleIncomingWebhook", "web.incoming_webhook.permissions.app_error", nil, "", http.StatusForbidden)
 	}
 
-	overrideUsername := req.Username
-	overrideIconUrl := req.IconURL
+	overrideUsername := hook.PostUsername
```

Comment:
This is one area of code where I have not added unit test coverage. This method as a whole is not unit tested, and to properly unit test these changes I'd need to do some refactoring in `CreateWebhookPost` to at least return the posted hooks, or build out some test helpers to query the database for the created posts. I felt this to be out of scope for the nature of this PR.

Reply:
After responding to @enahum's feedback below, I (re-)discovered the unit tests in https://github.com/mattermost/mattermost-server/blob/master/api4/webhook_test.go that need to be updated as part of this change. I think I had focussed too much on the `app/` changes (for which I wrote the necessary new unit tests) since I didn't have to edit anything in `api/` or `api4/` explicitly. I'll rectify this along with the other comment below. Does it makes sense to keep my https://github.com/mattermost/mattermost-server/pull/8002/files#diff-affca0a0eaa0a6e8f950a2d1b9f3582a changes as well or should I remove them in favour of updated `api4/` changes?

Reply:
I've updated the tests, keeping both. I don't yet have a full mental model of whose responsibility it is to test what (`api4` vs. `model`), so let me know if I've made any poor judgement calls.

Reply:
I think it makes sense to keep both 👍